### PR TITLE
de.rwth_aachen.ient.YUView

### PR DESCRIPTION
**Pull request by nolyn**

### CODE REVIEW

**Comment by eszlari** [comment]
The icons and desktop file are included in the upstream source, why don't you use them?

File: de.ient.YUView.desktop [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Type=Application
+Exec=YUView %F
+Name=YUView
+GenericName=YUV Video Player
+X-KDE-StartupNotify=true
+Icon=de.ient.YUView
+Terminal=false
+Categories=AudioVideo
+MimeType=application/vnd.ient.yuview.yuvfile
```
Comment (eszlari) [inline de.ient.YUView.desktop]: You will need a extra mime file to make this work, e.g.: https://github.com/flathub/com.jetbrains.PyCharm-Community/blob/master/pycharm.xml

File: de.ient.YUView.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+sdk: org.kde.Sdk
+command: YUView
+finish-args:
+  - --socket=wayland
+  - --socket=x11
+  - --share=ipc
+  - --device=dri
+  - --filesystem=host
+  - --share=network
+  - --env=PATH=/usr/bin:/app/bin:/app/bin/x86_64-linux
```
Comment (eszlari) [inline de.ient.YUView.yaml]: Why is this needed?

File: de.ient.YUView.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  - --env=QT_NO_FT_CACHE=1
+build-options:
+  env:
+    LD_LIBRARY_PATH: /app/lib:/app/lib64
```
Comment (eszlari) [inline de.ient.YUView.yaml]: Why is this needed?

File: de.ient.YUView.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        branch: deployment
+        url: https://github.com/IENT/YUView.git
+        commit: 64ccb50b85eed1db9d44b7f26c960b0807927f33
+
+  - name: libde265
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=release
+      - -DONLY_LIB=ON
+      - -DCMAKE_INSTALL_LIBDIR:PATH=/app/lib
```
Comment (eszlari) [inline de.ient.YUView.yaml]: This should not be needed.

File: de.ient.YUView.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        branch: internal
+        url: https://github.com/nolyn/libde265.git
+        commit: 6a59bc176cc53f7e422dcb0078cbcff3246ea56b
+
+  - name: desktopFileAndIcons
+    buildsystem: simple
+    sources:
+      - type: git
+        branch: internal
+        url: https://github.com/nolyn/libde265.git
```
Comment (eszlari) [inline de.ient.YUView.yaml]: Why do you include this a second time?

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (eszlari) [inline flathub.json]: Why?

**Comment by eszlari** [comment]
You are missing a appdata.xml file, e.g.:
https://github.com/flathub/io.github.OpenToonz/blob/master/io.github.OpenToonz.appdata.xml

File: de.ient.YUView.yaml (hunk 4 above)
Comment (nolyn) [reply de.ient.YUView.yaml]: I can remove 
`
      - -DCMAKE_BUILD_TYPE=release
`

`
       - -DONLY_LIB=ON 
`
is required in order to only build the library and not the example programs.

If I remove
`
       - -DCMAKE_INSTALL_LIBDIR:PATH=/app/lib 
`
I get
`
CMake Error at libde265/CMakeLists.txt:89 (INSTALL):
  install TARGETS given no LIBRARY DESTINATION for shared library target
  "de265".
`
The related part in the Cmake is
```
   INSTALL(TARGETS
     #libaries
     ${LIBDE265_LIBRARY_NAME}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   )
```
Should  CMAKE_INSTALL_LIBDIR be set automatically by flatpak?

File: de.ient.YUView.yaml (hunk 3 above)
Comment (nolyn) [reply de.ient.YUView.yaml]:
```
- --env=QT_NO_FT_CACHE=1
```
is a workaround for a bug in QT which causes the application to crash sporadically. The rest can be removed, thanks.

File: de.ient.YUView.yaml (hunk 2 above)
Comment (nolyn) [reply de.ient.YUView.yaml]: Can be removed, thanks!

File: de.ient.YUView.yaml (hunk 5 above)
Comment (nolyn) [reply de.ient.YUView.yaml]: copy paste error, I removed it

File: flathub.json (hunk 6 above)
Comment (nolyn) [reply flathub.json]: I only have 64bit systems to test on.

File: de.ient.YUView.desktop (hunk 1 above)
Comment (nolyn) [reply de.ient.YUView.desktop]: Can I remove the MimeType field and just create the xml as in pycharm? pycharm also doesn't have a MimeType in the desktop entry

Comment (nolyn) [reply de.ient.YUView.desktop]: how does it derive the correct app. in the pycharm xml
```
application/PyCharm
```
but in the corresponding desktop file
```
Name=PyCharm Community Edition
GenericName=PyCharm Community Edition - The intelligent Python IDE
Exec=pycharm-desktop
```
souldn't those match?

**Comment by nolyn** [comment]
I added the appdata, and fixed the other issues except, the mimetype, for which I have to do some further reading

**Comment by nolyn** [comment]
I added the mime file, I think everything works now. 

Is it ok to add
```
  - --filesystem=host or home
```
?

This application is useless without access to files.

File: de.ient.YUView.yaml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        commit: 64ccb50b85eed1db9d44b7f26c960b0807927f33
+    build-commands:
+      - install -Dm644 images/IENT-YUView-32.png /app/share/icons/hicolor/32x32/apps/de.ient.YUView.png
+      - install -Dm644 images/IENT-YUView-64.png /app/share/icons/hicolor/64x64/apps/de.ient.YUView.png
+      - install -Dm644 images/IENT-YUView-128.png /app/share/icons/hicolor/128x128/apps/de.ient.YUView.png
+      - install -Dm644 images/IENT-YUView-256.png /app/share/icons/hicolor/256x256/apps/de.ient.YUView.png
+      - install -Dm644 images/IENT-YUView-512.png /app/share/icons/hicolor/512x512/apps/de.ient.YUView.png
+      - install -Dm644 images/IENT-YUView-1024.png /app/share/icons/hicolor/1024x1024/apps/de.ient.YUView.png
+
+  - name: libde265
```
Comment (eszlari) [inline de.ient.YUView.yaml]: If YUView depends on this library, then have to move it up in the manifest.

File: de.ient.YUView.yaml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  - --socket=x11
+  - --share=ipc
+  - --env=QT_NO_FT_CACHE=1
+
+modules:
+  - name: YUView
+    buildsystem: qmake
+    sources:
+      - type: git
+        branch: deployment
```
Comment (eszlari) [inline de.ient.YUView.yaml]: Since you seems to be an upstream contributor, it would be nice if you could add git tag / make a proper release. Flathub is meant be used for hosting stable releases of software.

**Comment by eszlari** [comment]
> This application is useless without access to files.

If you tested it without `--filesystem` and it doesn't work, then it's okay to add it. Normally, the file chooser portal (http://docs.flatpak.org/en/latest/sandbox-permissions.html) should take care of it.

**Comment by eszlari** [comment]
Since you are an upstream contributor, add desktop, appdata, mime and icon files to the upstream source tree. For qmake integration see e.g.:

https://github.com/mltframework/shotcut/blob/master/src/src.pro#L355

File: de.ient.YUView.yaml (hunk 7 above)
Comment (nolyn) [reply de.ient.YUView.yaml]: that library is loaded at runtime, so it's not required for building. I've been testing all these changes locally with flatpak-builder and it works. I could still move it up, but I don't think it will make a difference.

File: de.ient.YUView.yaml (hunk 8 above)
Comment (nolyn) [reply de.ient.YUView.yaml]: We're currently about to make a new stable release, but it's not done yet. Probably another week or two. While we're doing this we're also moving to flatpak. I'll set the tag, once we have the stable release.

File: de.ient.YUView.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+app-id: de.ient.YUView
+runtime: org.kde.Platform
+runtime-version: 5.11
```
[the remark this answers is not in the document]
Comment (nolyn) [reply de.ient.YUView.yaml]: changed

**Comment by barthalion** [comment]
Fails to build on anything except x86_64: https://flathub.org/builds/#/builders/2/builds/3565

**Comment by nedrichards** [comment]
bot, build de.ient.YUView

File: de.ient.YUView.yaml [hunk 10]
```diff
@@ -0,0 +1,31 @@
+app-id: de.ient.YUView
```
Comment (barthalion) [inline de.ient.YUView.yaml]: It does not seem to load a website here. Just to make sure, do you own this domain?

**Comment by barthalion** [comment]
I think it's good to merge. Should someone additional from upstream team get push access?

File: de.ient.YUView.yaml (hunk 10 above)
Comment (nolyn) [reply de.ient.YUView.yaml]: Nope. The docs say
`
Flatpak identifies each application and runtime using a unique three-part identifier, such as com.company.App
`
I didn't realize this should correspond to a website. Does it have to? The corresponding websites would be https://github.com/IENT/YUView and http://www.ient.rwth-aachen.de. The latter is our institutes website and doesn't mention this software anywhere.

**Comment by nolyn** [comment]
Great to hear that.  If you can give access to the whole team, that would be https://github.com/IENT/. Else I'll take care of it for now.

File: de.ient.YUView.yaml (hunk 10 above)
Comment (barthalion) [reply de.ient.YUView.yaml]: I talked with @TingPing and in general we prefer something you control, with exceptions for git hosting services like GitHub or GitLab. The domain doesn't really need to mention the project, so whatever suits you best.

Comment (nolyn) [reply de.ient.YUView.yaml]: can it have more than 3 parts? i found these others:
https://github.com/flathub/com.github.bajoja.indicator-kdeconnect
https://github.com/flathub/com.github.marktext.marktext

also there does not seem to be a website for example at https://marktext.marktext.github.com/

I think we could do de.ient.rwth-aachen.YUView and de.ient.rwth-aachen.RDPlot for the other tool I'm still trying to package

Comment (barthalion) [reply de.ient.YUView.yaml]: Yes, 3 is minimum, but it can have more. Looks good; let's change it, run final test build and merge it!

Comment (nolyn) [reply de.ient.YUView.yaml]: alright! should I create a new branch with that name or do the changes in this one?

Comment (nolyn) [reply de.ient.YUView.yaml]: wait, should it be 
de.ient.rwth-aachen.YUView 
or 
de.rwth-aachen.ient.YUView
?

Comment (barthalion) [reply de.ient.YUView.yaml]: de.rwth-aachenient.YUView

Just rename the files and change the appid where needed, branch can stay the same. Thanks!

Comment (nolyn) [reply de.ient.YUView.yaml]: now I have this problem:
`
error: 'de.rwth-aachen.ient.YUView' is not a valid application name: Only last name segment can contain -
`

Comment (barthalion) [reply de.ient.YUView.yaml]: Right, you'll have to replace the dash with an undescore.

Comment (nolyn) [reply de.ient.YUView.yaml]: should be finished now :)

**Comment by nedrichards** [comment]
bot, build de.rwth_aachen.ient.YUView

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/de.rwth_aachen.ient.YUView

@nolyn, @fjaeger and @christianfeldmann, please make sure to accept the invitation that you should have received via e-mail to get push access to the repo above.

Thanks!
